src/dos/dos_locale.cpp

```diff
@@ -3496,15 +1543,42 @@ void DOS_Locale_Init(Section* sec)
 void DOS_Locale_AddMessages()
 {
 	MSG_Add("DOSBOX_HELP_LIST_COUNTRIES_1",
-	        "List of country codes (mostly same as telephone call codes)\n"
-	        "-----------------------------------------------------------");
+	        "List of available country codes (mostly same as telephone call codes)");
 	MSG_Add("DOSBOX_HELP_LIST_COUNTRIES_2",
-	        "The above numeric country codes can be used exactly as listed\n"
-	        "in the 'country' config setting.");
+	        "The above codes can be used in the 'country' config setting.");
+
+	MSG_Add("DOSBOX_HELP_LIST_KEYBOARD_LAYOUTS_1",
+	        "List of available keyboard layout codes");
+	MSG_Add("DOSBOX_HELP_LIST_KEYBOARD_LAYOUTS_2",
+	        "The above codes can be used in the 'keyboard_layout' config setting.");
 
 	// Add strings with country names
-	for (auto it = CountryData.begin(); it != CountryData.end(); ++it) {
+	for (auto it = LocaleData::CountryInfo.begin();
+	     it != LocaleData::CountryInfo.end();
+	     ++it) {
 		MSG_Add(it->second.GetMsgName().c_str(),
 		        it->second.country_name.c_str());
 	}
+
+	// Add strings with keyboard layout names
+	for (const auto& entry : LocaleData::KeyboardLayoutInfo) {
+		MSG_Add(entry.GetMsgName().c_str(), entry.layout_name.c_str());
+	}
+
+	MSG_Add("SCRIPT_LATIN",    "Latin");
+	MSG_Add("SCRIPT_ARABIC",   "Arabic");
+	MSG_Add("SCRIPT_ARMENIAN", "Armenian");
+	MSG_Add("SCRIPT_CHEROKEE", "Cherokee");
+	MSG_Add("SCRIPT_CYRILLIC", "Cyrillic");
+	MSG_Add("SCRIPT_GEORGIAN", "Georgian");
+	MSG_Add("SCRIPT_GREEK",    "Greek");
+	MSG_Add("SCRIPT_HEBREW",   "Hebrew");
+	MSG_Add("SCRIPT_THAI",     "Thai");
+
+	MSG_Add("KEYBOARD_SHORTCUT_SCRIPT_1", "LeftAlt+LeftShift");
+	MSG_Add("KEYBOARD_SHORTCUT_SCRIPT_2", "LeftAlt+RightShift");
+	MSG_Add("KEYBOARD_SHORTCUT_SCRIPT_3", "LeftAlt+LeftCtrl");
```

Comment:
Note to self: we can't hardcode the _Alt_ key names, as the Macintosh keyboards do not have them, they have _Command_ and _Option_ keys instead. To be checked how this is handled in the `INTRO` command or the startup banner.

Reply:
Problem fixed.